arm-service-map/2015-11-01-preview/swagger/arm-service-map.json

```diff
@@ -0,0 +1,2508 @@
+        "description": "Returns summary information about the machines in the workspace.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "startTime",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "format": "date-time",
+            "description": "UTC date and time specifying the start time of an interval. When not specified the service uses DateTime.UtcNow - 10m"
+          },
+          {
+            "name": "endTime",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "format": "date-time",
+            "description": "UTC date and time specifying the end time of an interval. When not specified the service uses DateTime.UtcNow"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/MachinesSummary"
+            }
+          },
+          "default": {
+            "description": "An error occurred while processing the request. See the error.code parameter to identify the specific error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "description": "Base for all ARM-adressable resources",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource identifier."
+        },
+        "type": {
+          "type": "string",
+          "description": "Resource type."
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Resource ETAG."
+        },
+        "kind": {
+          "type": "string",
+          "description": "Additional resource type qualifier."
+        }
+      },
+      "x-ms-external": true,
+      "x-ms-azure-resource": true,
+      "required": [
+        "id",
+        "type",
+        "name"
+      ]
+    },
+    "ResourceReference": {
+      "description": "Represents a reference to another resource.",
+      "discriminator": "kind",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource URI."
+        },
+        "type": {
+          "type": "string",
+          "description": "Additional resource type qualifier.  Always present when retrieving data from the service.  Optional when sending data to the service."
```

Comment:
The type field is meant to be supplied by the API when returning a reference to another resource. It makes the response more descriptive as it avoids the need to have the client parse the resulting id to determine the resource type. The type is not required when sending to the API. Swagger does not provide a way to express properly the distinction between inbound and outbound write-ability of members other than defining different types, which is rather clumsy.

If the user sends an id and type that are not compatible, this will result in Bad Request.